tests/subsys/pm/power_domain/src/main.c

```diff
@@ -279,5 +279,11 @@ ZTEST(power_domain_1cpu, test_power_domain_device_balanced)
 	zassert_equal(state, PM_DEVICE_STATE_ACTIVE);
 }
 
+ZTEST(power_domain_1cpu, test_on_power_domain)
+{
+	zassert_true(device_is_ready(domain), "Device is not ready!");
+	zassert_false(pm_device_on_power_domain(domain), "Device is on a switchable power domain.");
```

Comment:
While it is ok to have nested domains ... what you probably want to check here is if `deva`, `devb` and `devc` are in the domain `domain`. e.g:
```
zassert_true(pm_device_on_power_domain(deva), "deva is not in the power domain.");
(void)pm_device_power_domain_remove(devc, domain);
zassert_false(pm_device_on_power_domain(devc), "devc is in the power domain.");
(void)pm_device_power_domain_add(devc, domain);
zassert_true(pm_device_on_power_domain(devc), "devc is not in the power domain.");
```